pyccel/parser/semantic.py

```diff
@@ -2967,6 +2967,26 @@ def _visit_FunctionDef(self, expr, **settings):
         if len(not_used) >= 1:
             errors.report(UNUSED_DECORATORS, symbol=', '.join(not_used), severity='warning')
 
+        # Check if an argument is modified
+        
+        def check_if_const(arguments, code_block):
+            lines = code_block.body
+            for line in lines:
+                if isinstance(line, For) :
+                    check_if_const(arguments, line.body)
+                    
+                if isinstance(line, Assign):
+                    for arg in arguments:
+                        if (isinstance(line.lhs, PyccelSymbol) and line.lhs == arg.var)\
```

Comment:
This should occur after the code has been visited in the semantic stage. If that is the case then there won't be any `PyccelSymbol`s left, they should have all been converted to `Variable`s